File: ebike-device-paas-go/internal/pkg/logger/logger.go
package logger

import (
	"os"
	"strings"

	"go.uber.org/zap"
	"go.uber.org/zap/zapcore"
)

// Log is the global structured logger. It defaults to a no-op until Init is called.
var Log *zap.Logger = zap.NewNop()

// Init configures the global logger at the given level (debug/info/warn/error).
func Init(levelStr string) {
	var level zapcore.Level
	switch strings.ToLower(levelStr) {
	case "debug":
		level = zapcore.DebugLevel
	case "warn":
		level = zapcore.WarnLevel
	case "error":
		level = zapcore.ErrorLevel
	default:
		level = zapcore.InfoLevel
	}

	encoderCfg := zap.NewProductionEncoderConfig()
	encoderCfg.TimeKey = "timestamp"
	encoderCfg.EncodeTime = zapcore.ISO8601TimeEncoder

	core := zapcore.NewCore(
		zapcore.NewJSONEncoder(encoderCfg),
		zapcore.AddSync(os.Stdout),
		level,
	)
	Log = zap.New(core, zap.AddCaller())
}

File: ebike-UniApp/src/shared/logger.ts
type Level = 'debug' | 'info' | 'warn' | 'error'

function emit(level: Level, ...args: unknown[]) {
  const prefix = `[luopingtech][${level}]`
  // eslint-disable-next-line no-console
  console[level === 'debug' ? 'log' : level](prefix, ...args)
}

export const logger = {
  debug: (...args: unknown[]) => emit('debug', ...args),
  info: (...args: unknown[]) => emit('info', ...args),
  warn: (...args: unknown[]) => emit('warn', ...args),
  error: (...args: unknown[]) => emit('error', ...args),
}
